### ISSUE TITLE
fix(html-spec): add speculationrules to script type attribute

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3586]
## Describe the bug

The `invalid-attr` rule reports `<script type="speculationrules">` as an invalid MIME type. The Speculation Rules API is part of the HTML spec and `speculationrules` is a valid script type.

> https://html.spec.whatwg.org/multipage/scripting.html#attr-script-type
> https://wicg.github.io/nav-speculation/speculation-rules.html

## Code Example

```html
<script type="speculationrules">
{"prerender": [{"urls": ["/page"]}]}
</script>
```

## The current behavior

```
The "type" attribute expects the MIME Type format. Or, the "type" attribute expects either "module", "importmap"
```

## The expected behavior

No violations reported. `speculationrules` should be recognized as a valid script type alongside `module` and `importmap`.

## Impact

Causes ~12 false positives in the nu-html-checker compatibility benchmark.

## Affected package

`@markuplint/html-spec` — script element spec data
